data-prepper-plugins/saas-source-plugins/jira-source/src/main/java/org/opensearch/dataprepper/plugins/source/jira/JiraSourceConfig.java

```diff
@@ -0,0 +1,122 @@
+package org.opensearch.dataprepper.plugins.source.jira;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import jakarta.validation.constraints.Size;
+import lombok.AccessLevel;
+import lombok.Getter;
+import lombok.Setter;
+import org.opensearch.dataprepper.plugins.source.source_crawler.base.CrawlerSourceConfig;
+
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.opensearch.dataprepper.plugins.source.jira.utils.Constants.OAUTH2;
+
+@Getter
+public class JiraSourceConfig implements CrawlerSourceConfig {
+
+    private static final Duration DEFAULT_BACKOFF_MILLIS = Duration.ofMinutes(2);
+    /**
+     * This field aims at holding all the additional properties which are not specified above.
+     *
+     * <p>Needs to be a Map of String to Object to ensure that the value could be a nested structure
+     * as well.
+     */
+    @Setter(AccessLevel.NONE)
+    Map<String, Object> additionalProperties = new HashMap<>();
+    /**
```

Comment:
I think we have to write `@JsonDesription` for each of these

Reply:
I just verified KDS source object and I don't see any. Could you please point me to any example?

Reply:
For example - https://github.com/opensearch-project/data-prepper/blob/main/data-prepper-plugins/mutate-string-processors/src/main/java/org/opensearch/dataprepper/plugins/processor/mutatestring/SplitStringProcessorConfig.java. But it looks like only processors have this. Probably not required for Sources? Please check with George Chen